File: Herança - Subclasses de Personagem/aula4_heranca_PY/Personagem.py
class Personagem:
    def __init__(self):
        self._nome = None
        self._vida = None
        self._nivel = None
        self._inventario = []

    def pegar(self, item):
        self._inventario.append(item)

    @property
    def inventario(self):
        return self._inventario

    def ficha(self):
        resultado = "Nome: " + str(self.nome) + "\n" + \
        "Vida: " + str(self.vida) + "\n" + \
        "Nível: " + str(self.nivel) + "\n" + \
        "Inventário: "

        for item in self.inventario:
                resultado += "\n- " + item.descricao()
        return resultado
    
    @property
    def nome(self):
        return self._nome

    @nome.setter
    def nome(self, valor):
        if valor is None or valor == "":
            print("Erro de validação")
        else:
            self._nome = valor

    @property
    def vida(self):
        return self._vida
    
    @vida.setter
    def vida(self, valor):
        if 0 <= valor <= 100:
            self._vida = valor
        else:
            print("Erro de validação")

    @property
    def nivel(self):
        return self._nivel
    
    @nivel.setter
    def nivel(self, valor):
        if valor >= 1:
            self._nivel = valor 
        else:
            print("Erro de validação")

    def receber_dano(self, dano):
            self.vida -= dano

File: Herança - Subclasses de Personagem/aula4_heranca_JAVA/Personagem.java
package aula4_heranca_JAVA;
import java.util.ArrayList;

public class Personagem {
    private String nome;
    private int vida;
    private int nivel; 
    private ArrayList<Item> inventario;
    
public Personagem(String nome, int vida, int nivel){
    setNome(nome);
    setVida(vida);
    setNivel(nivel);
    inventario = new ArrayList<>();
}
public String getNome(){
   return nome;
}
public int getVida(){
    return vida;
}
public int getNivel(){
    return nivel;
}
public void setNome(String nome){
    if(nome == null || nome.isEmpty()){
        System.out.println("ERRO DE VALIDAÇAO!");
    } else{
        this.nome = nome;
        }
}
public void setVida(int v){
    if(v >= 0 && v <= 100){
        this.vida = v; 
    } else {
        System.out.println("ERRO DE VALIDAÇAO!");
    }
}
public void setNivel(int n){
    if(n < 1){
        System.out.println("ERRO DE VALIDAÇAO!");
    } else {
        this.nivel = n;
    }
}
public String ficha() {
    String resultado = "Nome: " + nome + "\n" + 
    "Vida: " + vida +"\n" +
    "Nível: " + nivel + "\n" + 
    "Inventário: "; 
    for (Item item : inventario) {
    resultado += "\n- " + item.descricao();
    }
    return resultado;
}
public void receberDano(int dano){
    vida -= dano;
}
public void pegar(Item item) {
    inventario.add(item);
}
}